src/snowflake/snowpark/dataframe.py

```diff
@@ -267,6 +278,7 @@ def _disambiguate(
         for n in lhs_names
         if n in set(rhs_names) and n not in normalized_using_columns
     ]
+    all_names = [unquote_if_quoted(n) for n in lhs_names + rhs_names]
 
```

Comment:
this is basically a snowpark python fix, right? AST does not seem to be affected?

Love the additional test.